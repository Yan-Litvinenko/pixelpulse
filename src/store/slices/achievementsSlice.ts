import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { customFetch } from '../../utils/customFetch';
import type { IAchieve } from '../../interfaces/interface.achievements';

type AchievementsSlice = {
    amountAchievements: null | number;
    amountAchieved: null | number;
    percent: null | number;
    achievements: IAchieve[] | null;
    loading: boolean;
    error: null | string;
};

const initialState: AchievementsSlice = {
    amountAchieved: null,
    amountAchievements: null,
    percent: null,
    achievements: null,
    loading: false,
    error: null,
};

const fetchAchievements = createAsyncThunk('achievements/fetchAchievements', async (): Promise<IAchieve[]> => {
    const resolveAchievements = await customFetch<IAchieve[]>('/api/getAchievements');
    return resolveAchievements;
});

const achievedCount = (allAchievements: IAchieve[]): number =>
    allAchievements.reduce((acc, { status }) => acc + (status === 'achieved' ? 1 : 0), 0);

const achievementsSlice = createSlice({
    name: 'achievements',
    initialState,
    reducers: {},
    extraReducers(builder) {
        builder
            .addCase(fetchAchievements.fulfilled, (state, action) => {
                const achievements: IAchieve[] = action.payload;
                const amountAchieved = achievedCount(action.payload);
                const amountAchievements = action.payload.length;
                const percent = (amountAchieved * 100) / amountAchievements;

                state.achievements = achievements;
                state.amountAchieved = amountAchieved;
                state.amountAchievements = amountAchievements;
                state.error = null;
                state.loading = false;
                state.percent = percent;
            })
            .addCase(fetchAchievements.pending, (state) => {
                state.error = null;
                state.loading = true;
            })
            .addCase(fetchAchievements.rejected, (state, action) => {
                state.error = action.error.message || 'Error achievements loading';
                state.loading = false;
            });
    },
});

export { achievementsSlice, fetchAchievements };
export type { AchievementsSlice };
