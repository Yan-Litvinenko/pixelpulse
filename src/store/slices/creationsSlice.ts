import projects from '../../assets/json/projects.json';
import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import type { IProject } from '../../interfaces/interface.creations';

type CreationsSlice = {
    readonly defaultProject: IProject;
    readonly projects: IProject[];
    targetProject: number;
    targetImage: number;
    projectImages: string[];
};

const initialState: CreationsSlice = {
    defaultProject: {
        about: [
            'The site design is made in the style of a computer game. It describes my achievements and skills, information about my life and goals.',
        ],
        brief: 'reflects growth in educational or professional achievements',
        images: [''],
        link: '',
        name: 'pixelpulse',
        technologies: ['ts', 'react', 'redux', 'nodeJS', 'mysql'],
    },
    targetProject: 0,
    targetImage: 0,
    projects: projects as IProject[],
    projectImages: [],
};

const creationsSlice = createSlice({
    name: 'creations',
    initialState,
    reducers: {
        setProjectImages(state, action: PayloadAction<string[]>) {
            state.projectImages = action.payload;
        },
        setTargetProject(state, action: PayloadAction<number>) {
            state.targetProject = action.payload;
        },
        setTargetImage(state, action: PayloadAction<number>) {
            state.targetImage = action.payload;
        },
    },
});

export const { setProjectImages, setTargetImage, setTargetProject } = creationsSlice.actions;
export { creationsSlice };
