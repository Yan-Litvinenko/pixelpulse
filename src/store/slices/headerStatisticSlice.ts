import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { customFetch } from '../../utils/customFetch';
import type { IAddCoinResult } from '../../interfaces/interface';

type HeaderStatistic = {
    statistic: {
        level: number | null;
        coins: number | null;
        addStatus: boolean;
    };
    loading: boolean;
    error: string | null;
};

const initialState: HeaderStatistic = {
    statistic: {
        level: null,
        coins: null,
        addStatus: false,
    },
    loading: false,
    error: null,
};

const fetchHeaderStatistic = createAsyncThunk('headerStatistic/fetchHeaderStatistic', async () => {
    const resolveLevel: number = await customFetch<number>('/api/getLevel');
    const resolveCoins: number = await customFetch<number>('/api/getCoins');
    const resolveAddStatus: boolean = await customFetch<boolean>('/api/getStatusAddToday');

    return {
        level: resolveLevel,
        coins: resolveCoins,
        addStatus: resolveAddStatus,
    };
});

const fetchAddCoin = createAsyncThunk<
    IAddCoinResult,
    React.MouseEvent<HTMLButtonElement, MouseEvent>,
    { rejectValue: string }
>('headerStatistic/fetchAddCoin', async (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    event.preventDefault();

    const response: Response = await fetch('/addCoin', { method: 'POST' });
    const updateStatistic: IAddCoinResult = await response.json();

    return new Promise((resolve) => setTimeout(() => resolve(updateStatistic), 500));
});

const headerStatisticSlice = createSlice({
    name: 'headerStatistic',
    initialState,
    reducers: {},
    extraReducers(builder) {
        builder
            .addCase(fetchHeaderStatistic.fulfilled, (state, action) => {
                state.error = null;
                state.loading = false;

                state.statistic.coins = action.payload.coins;
                state.statistic.level = action.payload.level;
                state.statistic.addStatus = action.payload.addStatus;
            })
            .addCase(fetchHeaderStatistic.pending, (state) => {
                state.loading = true;
                state.error = null;
            })
            .addCase(fetchHeaderStatistic.rejected, (state, action) => {
                state.loading = false;
                state.error = action.error.message || 'Error loading statistic';
            })

            .addCase(fetchAddCoin.fulfilled, (state, action) => {
                state.error = null;
                state.loading = false;

                if (action.payload) {
                    state.statistic.level = action.payload.level;
                    state.statistic.coins = action.payload.coins;
                    state.statistic.addStatus = action.payload.addStatus;
                }
            })
            .addCase(fetchAddCoin.pending, (state) => {
                state.loading = true;
                state.error = null;
            })
            .addCase(fetchAddCoin.rejected, (state, action) => {
                state.loading = false;
                state.error = action.error.message || 'Error added coins';
            });
    },
});

export { headerStatisticSlice, fetchAddCoin, fetchHeaderStatistic };
