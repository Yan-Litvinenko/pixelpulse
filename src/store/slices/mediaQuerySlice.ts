import { createSlice } from '@reduxjs/toolkit';

type MediaQueryState = {
    readonly SMALL: number;
    readonly MEDIUM: number;
    isSmall: boolean;
    isMedium: boolean;
};

type SetMediaQueryState = Pick<MediaQueryState, 'isSmall' | 'isMedium'>;

const initialState: MediaQueryState = {
    SMALL: 768,
    MEDIUM: 1200,
    isSmall: false,
    isMedium: false,
};

const mediaQuerySlice = createSlice({
    name: 'mediaQuery',
    initialState,
    reducers: {
        setMediaState(state, action) {
            state.isSmall = (action.payload as SetMediaQueryState).isSmall;
            state.isMedium = (action.payload as SetMediaQueryState).isMedium;
        },
    },
});

export const { setMediaState } = mediaQuerySlice.actions;
export type { MediaQueryState };
export { mediaQuerySlice };
