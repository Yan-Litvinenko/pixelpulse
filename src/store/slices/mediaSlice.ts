import { createSlice } from '@reduxjs/toolkit';

type MediaState = {
    readonly SMALL: number;
    readonly MEDIUM: number;
    isSmall: boolean;
    isMedium: boolean;
};

type SetMediaState = Pick<MediaState, 'isSmall' | 'isMedium'>;

const initialState: MediaState = {
    SMALL: 768,
    MEDIUM: 1200,
    isSmall: false,
    isMedium: false,
};

const mediaSlice = createSlice({
    name: 'mediaQuery',
    initialState,
    reducers: {
        setMediaState(state, action) {
            state.isSmall = (action.payload as SetMediaState).isSmall;
            state.isMedium = (action.payload as SetMediaState).isMedium;
        },
    },
});

export const { setMediaState } = mediaSlice.actions;
export { mediaSlice };
