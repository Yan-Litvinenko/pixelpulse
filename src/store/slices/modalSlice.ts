import { createSlice } from '@reduxjs/toolkit';
import { scroll } from '../../classes/Scroll';
import type { PayloadAction } from '@reduxjs/toolkit';
import type { AppThunk } from '../store';

type Modal = 'social' | 'availability' | 'creations' | 'credits' | 'settings' | 'challenge' | 'navigationMobile';
type SetValueAndKeyReduce = {
    key: Modal;
    value: boolean;
};
type SetKeyReduce = Pick<SetValueAndKeyReduce, 'key'>;
type ModalSlice = {
    readonly TRANSITION_TIME: number;
    stateModal: Record<Modal, boolean>;
    stateForm: Record<Modal, boolean>;
    delay: Record<Modal, boolean>;
};

const initialState: ModalSlice = {
    TRANSITION_TIME: 1500,
    stateModal: {
        availability: false,
        social: false,
        creations: false,
        credits: false,
        settings: false,
        challenge: false,
        navigationMobile: false,
    },
    stateForm: {
        availability: false,
        social: false,
        creations: false,
        credits: false,
        settings: false,
        challenge: false,
        navigationMobile: false,
    },
    delay: {
        availability: false,
        social: false,
        creations: false,
        credits: false,
        settings: false,
        challenge: false,
        navigationMobile: false,
    },
};

const modalSlice = createSlice({
    name: 'availability',
    initialState,
    reducers: {
        setDelay(state, action: PayloadAction<SetValueAndKeyReduce>) {
            state.delay[action.payload.key] = action.payload.value;
        },
        setModalStateForm(state, action: PayloadAction<SetValueAndKeyReduce>) {
            state.stateForm[action.payload.key] = action.payload.value;
        },
        open(state, action: PayloadAction<SetKeyReduce>) {
            state.stateModal[action.payload.key] = true;
        },
        close(state, action: PayloadAction<SetKeyReduce>) {
            state.stateModal[action.payload.key] = false;
            state.stateForm[action.payload.key] = false;
            state.delay[action.payload.key] = false;
        },
    },
});

const { setDelay, setModalStateForm, open, close } = modalSlice.actions;

const modalOpenHandler =
    (props: SetKeyReduce): AppThunk =>
    (dispatch, getState) => {
        const { TRANSITION_TIME } = getState().modal;
        const { isMedium } = getState().mediaQuery;
        const { key } = props;

        const openHelper = (): void => {
            window.removeEventListener('scrollend', openHelper);

            if (isMedium) {
                dispatch(setDelay({ key, value: true }));
            } else {
                setTimeout(() => dispatch(setDelay({ key, value: true })), TRANSITION_TIME);
            }

            scroll.off();
            dispatch(open({ key }));
        };

        if (window.scrollY !== 0) {
            window.addEventListener('scrollend', openHelper);
            scroll.moveTop();
            return;
        }

        openHelper();
    };

const modalCloseHandler =
    (props: SetKeyReduce): AppThunk =>
    (dispatch, getState) => {
        const { key } = props;
        const { delay, stateForm } = getState().modal;

        if (!delay[key] || stateForm[key]) return;

        dispatch(close({ key }));
        scroll.on();
    };

export { modalSlice, modalOpenHandler, setModalStateForm, modalCloseHandler };
export type { Modal };
