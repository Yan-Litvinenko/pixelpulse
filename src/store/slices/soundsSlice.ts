import linkSoundClick from '../../assets/audio/click.ogg';
import linkSoundModal from '../../assets/audio/modal.mp3';
import linkGlitchEffect from '../../assets/audio/glitch.mp3';
import { createSlice } from '@reduxjs/toolkit';
import { getValueToLocalStorage } from '../../utils/getValueToLocalStorage';

type SoundsSlice = {
    readonly linkGlitchEffect: string;
    readonly linkSoundClick: string;
    readonly linkSoundModal: string;
    soundsState: boolean;
    stateClickTrigger: boolean;
    stateGlitchTrigger: boolean;
    stateModalTrigger: boolean;
};

const initialState: SoundsSlice = {
    linkGlitchEffect,
    linkSoundClick,
    linkSoundModal,
    soundsState: getValueToLocalStorage<boolean>('pixelpulse-sounds-state', true),
    stateClickTrigger: false,
    stateGlitchTrigger: false,
    stateModalTrigger: false,
};

const soundsSlice = createSlice({
    name: 'sounds',
    initialState,
    reducers: {
        setStateSounds(state) {
            state.soundsState = !state.soundsState;
        },
        soundsClickTrigger(state) {
            state.stateClickTrigger = !state.stateClickTrigger;
        },
        soundsModalTrigger(state) {
            state.stateModalTrigger = !state.stateModalTrigger;
        },
        soundsGlitchTrigger(state) {
            state.stateGlitchTrigger = !state.stateGlitchTrigger;
        },
    },
});

export const { setStateSounds, soundsModalTrigger, soundsClickTrigger, soundsGlitchTrigger } = soundsSlice.actions;
export type { SoundsSlice };
export { soundsSlice };
