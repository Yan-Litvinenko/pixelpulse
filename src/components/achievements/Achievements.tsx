import React from 'react';
import styles from './Achievements.module.scss';
import { AchievementsBlock } from '../achievementsBlock/AchievementsBlock';
import { AchievementsError } from '../achievementsError/AchievementsError';
import { achievementsFilter } from '../../utils/achievementsFilter';
import { AchievementsProgress } from '../achievementsProgress/AchievementsProgress';
import { achievementsSelector } from '../../store/selectors/selectors';
import { achievementsSort } from '../../utils/achievementsSort';
import { AchievementsToggle } from '../achievementsToggle/AchievementsToggle';
import { soundsClickTrigger } from '../../store/slices/soundsSlice';
import { Triangle } from 'react-loader-spinner';
import { useDispatch, useSelector } from 'react-redux';
import type { AppDispatch } from '../../store/store';
import type { ToggleStatus } from '../../interfaces/interface.achievements';

const Achievements = (): React.JSX.Element => {
    const dispatch = useDispatch<AppDispatch>();
    const { achievements, loading, error } = useSelector(achievementsSelector);
    const [filterStatus, setFilterStatus] = React.useState<ToggleStatus>('all');

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
        setFilterStatus(event.target.value as ToggleStatus);
        dispatch(soundsClickTrigger());
    };

    return (
        <section className={styles.achievements}>
            <h2 className={styles.achievements__title}>{error ? 'Error achievements loading' : 'Achievements'}</h2>

            <div className={styles.achievements__content}>
                <AchievementsProgress />

                <div className={styles.achievements__achievements}>
                    {loading ? (
                        <Triangle
                            ariaLabel="triangle-loading"
                            color=""
                            height="120"
                            visible={true}
                            width="120"
                            wrapperClass={styles.loader}
                            wrapperStyle={{}}
                        />
                    ) : error ? (
                        <AchievementsError />
                    ) : (
                        <>
                            <AchievementsBlock
                                prefixForClassName={'achieved'}
                                achievements={achievementsSort(
                                    achievementsFilter(achievements || [], filterStatus, 'achieved'),
                                )}
                            />
                            <AchievementsBlock
                                prefixForClassName={'ongoing'}
                                achievements={achievementsSort(
                                    achievementsFilter(achievements || [], filterStatus, 'in progress'),
                                )}
                            />
                        </>
                    )}
                </div>

                <div className={styles.switchers}>
                    <AchievementsToggle
                        checked={filterStatus === 'all'}
                        id="all"
                        onChange={handleChange}
                        textContent="all"
                        value={'all'}
                    />
                    <AchievementsToggle
                        checked={filterStatus === 'achieved'}
                        id="achieved"
                        onChange={handleChange}
                        textContent="achieved"
                        value={'achieved'}
                    />
                    <AchievementsToggle
                        checked={filterStatus === 'inProgress'}
                        id="inProgress"
                        onChange={handleChange}
                        textContent="in progress"
                        value={'inProgress'}
                    />
                </div>
            </div>
        </section>
    );
};

export { Achievements };
