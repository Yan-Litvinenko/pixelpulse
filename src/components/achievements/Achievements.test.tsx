import React from 'react';
import { Achievements } from './Achievements';
import { render, screen, fireEvent } from '@testing-library/react';
import { soundsClickTrigger } from '../../store/slices/soundsSlice';
import * as reduxHooks from 'react-redux';

jest.mock('react-redux');

const mockUseSelector = jest.spyOn(reduxHooks, 'useSelector');
const mockUseDispatch = jest.spyOn(reduxHooks, 'useDispatch');

describe('Achievements component', (): void => {
    test('Achievements render', (): void => {
        mockUseDispatch.mockReturnValue(jest.fn());
        mockUseSelector.mockReturnValue({
            achievements: [
                {
                    id: 1,
                    date: 'ongoing',
                    description: 'Get level 80',
                    rarity: 'legendary',
                    status: 'in progress',
                    title: '80 level',
                },
            ],
            loading: false,
            error: false,
        });

        const component = render(<Achievements />);

        expect(screen.getByText('Achievements')).toBeInTheDocument();
        expect(component).toMatchSnapshot();
    });

    test('Achievements loading', (): void => {
        mockUseDispatch.mockReturnValue(jest.fn());
        mockUseSelector.mockReturnValue({ achievements: null, loading: true, error: false });

        render(<Achievements />);
        expect(screen.getByTestId('triangle-loading')).toBeInTheDocument();
    });

    test('Achievements error', (): void => {
        mockUseDispatch.mockReturnValue(jest.fn());
        mockUseSelector.mockReturnValue({ achievements: null, loading: false, error: true });

        render(<Achievements />);

        expect(screen.getByText(/Error achievements loading/i)).toBeInTheDocument();
        expect(screen.getByTestId('achievements-list-error')).toBeInTheDocument();
    });

    test('Should change the checkbox', (): void => {
        const dispatch = jest.fn();
        mockUseDispatch.mockReturnValue(dispatch);

        render(<Achievements />);

        const allCheckbox = screen.getByLabelText('all') as HTMLInputElement;
        const achievedCheckbox = screen.getByLabelText('achieved') as HTMLInputElement;
        const inProgressCheckbox = screen.getByLabelText('in progress') as HTMLInputElement;

        expect(allCheckbox.checked).toBe(true);
        fireEvent.click(achievedCheckbox);

        expect(achievedCheckbox.checked).toBe(true);
        fireEvent.click(inProgressCheckbox);

        expect(inProgressCheckbox.checked).toBe(true);
        expect(dispatch).toHaveBeenCalledTimes(2);
        expect(dispatch).toHaveBeenCalledWith(soundsClickTrigger());
    });
});
