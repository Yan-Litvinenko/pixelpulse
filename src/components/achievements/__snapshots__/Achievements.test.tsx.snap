// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievements component Achievements render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          Achievements
        </h2>
        <div>
          <section>
            <div>
              <svg
                data-testid="svg"
                height="148"
                viewBox="0 0 148 148"
                width="148"
              >
                <circle
                  cx="74"
                  cy="74"
                  data-testid="circle"
                  fill="transparent"
                  r="46"
                  stroke="current-color"
                  stroke-width="14"
                  style="stroke-dashoffset: NaN; stroke-dasharray: 289.02652413026095 289.02652413026095;"
                />
              </svg>
              <span>
                NaN
                /
                NaN
              </span>
              <span>
                progress
              </span>
            </div>
            <p>
              I have created a set of achievements for myself and I use this page to track them.
            </p>
            <p>
              If you want to give me a challenge and rate it, please feel free to submit it with the button below!
            </p>
            <button
              type="button"
            >
              Challenge me
            </button>
          </section>
          <div>
            <div
              data-testid="achievements-block"
            >
              <span>
                to be achieved:
              </span>
              <div>
                <article
                  class="undefined undefined undefined"
                >
                  <figure>
                    <img
                      alt="Rarity icon representing legendary"
                      src="test-file-stub"
                    />
                    <figcaption>
                      legendary
                    </figcaption>
                  </figure>
                  <div
                    class="undefined"
                  >
                    <div>
                      <h4>
                        80 level
                         
                      </h4>
                      <p>
                        Get level 80
                      </p>
                      <time
                        datetime="ongoing"
                      >
                        achieved:
                        ongoing
                      </time>
                    </div>
                  </div>
                </article>
              </div>
            </div>
          </div>
          <div>
            <label
              data-testid="test-achievements-label"
              for="all"
            >
              <input
                checked=""
                id="all"
                name="achievements"
                type="radio"
                value="all"
              />
              <div />
              <div>
                all
              </div>
            </label>
            <label
              data-testid="test-achievements-label"
              for="achieved"
            >
              <input
                id="achieved"
                name="achievements"
                type="radio"
                value="achieved"
              />
              <div />
              <div>
                achieved
              </div>
            </label>
            <label
              data-testid="test-achievements-label"
              for="inProgress"
            >
              <input
                id="inProgress"
                name="achievements"
                type="radio"
                value="inProgress"
              />
              <div />
              <div>
                in progress
              </div>
            </label>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <h2>
        Achievements
      </h2>
      <div>
        <section>
          <div>
            <svg
              data-testid="svg"
              height="148"
              viewBox="0 0 148 148"
              width="148"
            >
              <circle
                cx="74"
                cy="74"
                data-testid="circle"
                fill="transparent"
                r="46"
                stroke="current-color"
                stroke-width="14"
                style="stroke-dashoffset: NaN; stroke-dasharray: 289.02652413026095 289.02652413026095;"
              />
            </svg>
            <span>
              NaN
              /
              NaN
            </span>
            <span>
              progress
            </span>
          </div>
          <p>
            I have created a set of achievements for myself and I use this page to track them.
          </p>
          <p>
            If you want to give me a challenge and rate it, please feel free to submit it with the button below!
          </p>
          <button
            type="button"
          >
            Challenge me
          </button>
        </section>
        <div>
          <div
            data-testid="achievements-block"
          >
            <span>
              to be achieved:
            </span>
            <div>
              <article
                class="undefined undefined undefined"
              >
                <figure>
                  <img
                    alt="Rarity icon representing legendary"
                    src="test-file-stub"
                  />
                  <figcaption>
                    legendary
                  </figcaption>
                </figure>
                <div
                  class="undefined"
                >
                  <div>
                    <h4>
                      80 level
                       
                    </h4>
                    <p>
                      Get level 80
                    </p>
                    <time
                      datetime="ongoing"
                    >
                      achieved:
                      ongoing
                    </time>
                  </div>
                </div>
              </article>
            </div>
          </div>
        </div>
        <div>
          <label
            data-testid="test-achievements-label"
            for="all"
          >
            <input
              checked=""
              id="all"
              name="achievements"
              type="radio"
              value="all"
            />
            <div />
            <div>
              all
            </div>
          </label>
          <label
            data-testid="test-achievements-label"
            for="achieved"
          >
            <input
              id="achieved"
              name="achievements"
              type="radio"
              value="achieved"
            />
            <div />
            <div>
              achieved
            </div>
          </label>
          <label
            data-testid="test-achievements-label"
            for="inProgress"
          >
            <input
              id="inProgress"
              name="achievements"
              type="radio"
              value="inProgress"
            />
            <div />
            <div>
              in progress
            </div>
          </label>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
